docs/src/views/Architecture.tsx

```diff
@@ -0,0 +1,213 @@
+      Group, etc).
+    </p>
+    <p>
+      Library provides the set of accessibility behaviors and keyboard handlers for the components
+      and allows to create and inject your own, if it is necessary for some custom components. In a
+      most of the cases there should be enough to use the existing from the library.
+    </p>
+    <p>
+      In addition to behaviors and keyboard handlers, the toolkit also provides FocusZone component,
+      which can be used to add focus handling for the wrapped component. There is also FocusTrap
+      component, which allows to trap the focus inside some area, which is useful for such a
+      component as Dialog.
+    </p>
+
+    <Header as="h3">Theming</Header>
+    <p>
+      Component logic allows to inject and switch the component theme (e.g. Light, Dark or High
+      Contrast). To achieve this, we use CSS in JS technique, which is provided by{' '}
+      <a href="http://fela.js.org/" target="_blank">
+        Fela library
+      </a>.
+    </p>
+    <p>
+      Each component has its own set of themes defined. Component logic handles theme switching and
+      style overriding by providing ability to set <strong>variables</strong> or override component
+      (ot its children) styles using <code>styles</code> property.
+    </p>
+    <p>
+      Components are relaying on the theme provider, which has to wrap components to provide default
+      theme and theme switching. Providers can be nested, this is necessary in case if you need to
+      use different theme in children and parent components.
+    </p>
+
+    <Header as="h3">UIComponent</Header>
+    <p>
+      The <code>UIComponent</code> is the base component, which handles component default
+      properties, accessibility navigation and focus handling and renders the component.
+    </p>
+    <p>
+      We recommend to use <code>UIComponent</code> instead of <code>React.Component</code> when you
+      create your own primitive or composed component, to leverage ability of injecting
+      accessibility and theming, instead of implementing your own mechanisms for handling these
+      parts.
+    </p>
+
+    <Header as="h3">AutoControlledComponent</Header>
+    <p>
+      The <code>AutoControlledComponent</code> is the base component, which maintains a component
+      state and allows to set its initial state based on default properties and allows to override
+      state by property with the same name.
+    </p>
+
+    <Header as="h2">Component Library</Header>
+    <p>
+      The Component Library is a catalog of components, primitive and composed, which has predefined
+      behavior, themes and accessibility and can be customized using their properties or with
+      injecting additional themes and/or children components.
+    </p>
+
+    <Header as="h3">Components</Header>
+    <p>
+      Each component is inherited from <code>UIComponent</code> or{' '}
```

Comment:
this is the code added by pretier, or some similar tool, to add the space between the words when they are located on separate lines. So, it can't be removed :)